File: Soal2.js
function cek_palindrom(kalimat) {
    //Mengubah kalimat menjadi huruf kecil dan menghapus spasi
    kalimat = kalimat.toLowerCase().trim();
    //Mengecek apakah kalimat adalah palindrom
    return kalimat == kalimat.split('').reverse().join('');
}

function hasil_palindrom(kalimat) {
    if (cek_palindrom(kalimat)) {
        return "eureka!";
    }else {
        return "suka blyat";
    }
}

const contoh_kata = [
    "Angsa",
    "KataK",
    "kasur empuk",
    "Aku Suka Kamu",
    "Ibu ratna antar ubi",
    "Kasur ini rusak",
    "katak suka"
];

contoh_kata.forEach(kalimat => {
    console.log(`${kalimat} -> ${hasil_palindrom(kalimat)}`);
});

File: Soal2.py
def cek_palindrome(kalimat):
    # Mengubah kalimat menjadi huruf kecil serta menghapus spasi dan tanda baca
    kalimat_mentah = ''.join(huruf.lower() for huruf in kalimat if huruf.isalnum())
    
    if kalimat_mentah == kalimat_mentah[::-1]:
        return "eureeka!"
    else:
        return "suka blyat"
    
while True:
    inputan = input("Masukkan kata: ")
    if inputan.strip() != "":
        hasil = cek_palindrome(inputan)
        print(hasil)
        break
    else:
        print("Tolong input sesuatu")